### PR TITLE
Verify phone number

### PR DESCRIPTION
**Pull request by axeldelafosse**
# Why

We need to let users verify their phone numbers

# How

- Added a modal to verify a phone number
- Added a screen in the Settings to manage the phone number
- Added a check when claiming failed to verify the phone number to unlock more claims

# Test Plan

Run the apps and check if you can verify your phone number

Screenshot 2022-09-15 at 12 58 33
Screenshot 2022-09-15 at 12 58 31
Screenshot 2022-09-15 at 12 54 43
Screenshot 2022-09-15 at 12 54 38
Screenshot 2022-09-15 at 12 50 38
Screenshot 2022-09-15 at 12 50 33

### CODE REVIEW

File: packages/app/components/login/phone-number-picker.tsx [hunk 1]
```diff
@@ -79,53 +128,16 @@ export const PhoneNumberPicker = (props: PhoneNumberPickerProp) => {
...
         />
       </Modal>
+
       <LoginInputField
         key="login-phone-number-field"
         validationSchema={phoneNumberValidationSchema}
         label="Phone number"
         placeholder="Enter your phone number"
-        keyboardType="numeric"
+        keyboardType="phone-pad"
```
Comment (intergalacticspacehighway) [inline packages/app/components/login/phone-number-picker.tsx:137]: 💯

File: packages/app/components/settings/add-email.tsx [hunk 2]
```diff
@@ -27,25 +28,26 @@ export const AddEmailModal = () => {
 
   const submitEmail = useCallback(
     async (email: string) => {
+      rudder?.track("Button Clicked", {
+        name: "Login with email",
```
Comment (intergalacticspacehighway) [inline packages/app/components/settings/add-email.tsx:32]: should this be named `Login with email`?

File: packages/app/components/settings/add-email.tsx [hunk 3]
```diff
@@ -27,25 +28,26 @@ export const AddEmailModal = () => {
...
+
       try {
         const did = await magic.auth.loginWithMagicLink({ email });
         if (did) {
           await addEmail(email, did);
           router.pop();
         }
       } catch (error) {
-        console.log("Error:", error);
+        console.error(error);
```
Comment (intergalacticspacehighway) [inline packages/app/components/settings/add-email.tsx:42]: nit: we can use Logger.error(). Right now it doesn't do anything but in future we can avoid logging on prod and send it to sentry.

**Review by intergalacticspacehighway — APPROVED** [review]
LGTM ❇️

File: packages/app/components/settings/add-email.tsx (hunk 3 above)
Comment (alantoa) [reply packages/app/components/settings/add-email.tsx:42]: yeah agreed! use  `Logger.error` will be better!

**Comment by intergalacticspacehighway** [comment]
nit: design gap between button and text

IMG_F72FC0B31B37-1

**Comment by intergalacticspacehighway** [comment]
would be cool to initialise phone country [using this](https://docs.expo.dev/versions/latest/sdk/localization/#localizationregion)! (can be done is separate PR)
https://docs.expo.dev/versions/latest/sdk/localization/

File: packages/app/components/settings/add-email.tsx (hunk 2 above)
Comment (axeldelafosse) [reply packages/app/components/settings/add-email.tsx:32]: Nope, want to keep it consistent with the tracking for the login (the idea being to track someone using an email via Magic)

File: packages/app/components/settings/add-email.tsx (hunk 3 above)
Comment (axeldelafosse) [reply packages/app/components/settings/add-email.tsx:42]: OK good idea, let's switch all `console.error` to `Logger.error` in a separate PR

**Pull request merged**